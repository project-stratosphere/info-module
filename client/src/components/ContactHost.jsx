import React from 'react';
import {
  Wrapper,
  StyledButton,
  Modal,
  ModalContent,
  FacebookButton,
  GoogleButton,
  EmailButton,
  Line,
  Login,
} from './styles/ContactHost.styles';

export default class ContactHost extends React.Component {
  constructor(props) {
    super(props);
    this.handleModalDisplay = this.handleModalDisplay.bind(this);
    this.handleModalHide = this.handleModalHide.bind(this);
    this.state = {
      displayModal: 'none',
    };
  }

  handleModalDisplay() {
    this.setState({
      displayModal: 'flex',
    });
  }

  handleModalHide() {
    this.setState({
      displayModal: 'none',
    });
  }

  render() {
    return (
      <Wrapper>
        <StyledButton
          onClick={this.handleModalDisplay}
          className="modal-display"
        >
          Contact host
        </StyledButton>
        <Modal
          onClick={this.handleModalHide}
          displayModal={this.state.displayModal}
          className="modal-hide"
        >
          <ModalContent>
            <FacebookButton>
              Continue with Facebook
            </FacebookButton>
            <GoogleButton>
              Continue with Google
            </GoogleButton>
            <Line>
              <hr className="half-line" />
              or
              <hr className="half-line" />
            </Line>
            <EmailButton>
              Sign up with Email
            </EmailButton>
            <Line>
              <hr className="full-line" />
            </Line>
            <Login>
              Already have an Airbnb account?
              <span>
                Log in
              </span>
            </Login>
          </ModalContent>
        </Modal>
      </Wrapper>
    );
  }
}
