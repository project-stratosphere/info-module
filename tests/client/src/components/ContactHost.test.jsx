import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import ContactHost from '../../../../client/src/components/ContactHost';

Enzyme.configure({ adapter: new Adapter() });

describe('ContactHost', () => {
  it('should render correctly', () => {
    const output = Enzyme.shallow(<ContactHost />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should change state on modalDisplay and modalHide', () => {
    const output = Enzyme.shallow(<ContactHost />);
    expect(output.state().displayModal).toEqual('none');
    output.find('.modal-display').simulate('click');
    expect(output.state().displayModal).toEqual('flex');
    output.find('.modal-hide').simulate('click');
    expect(output.state().displayModal).toEqual('none');
  });
});
