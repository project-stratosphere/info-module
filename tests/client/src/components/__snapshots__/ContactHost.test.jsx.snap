// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactHost should render correctly 1`] = `
<styled.div>
  <styled.button
    className="modal-display"
    onClick={[Function]}
  >
    Contact host
  </styled.button>
  <styled.div
    className="modal-hide"
    displayModal="none"
    onClick={[Function]}
  >
    <styled.div>
      <styled.div>
        Continue with Facebook
      </styled.div>
      <styled.div>
        Continue with Google
      </styled.div>
      <styled.div>
        <hr
          className="half-line"
        />
        or
        <hr
          className="half-line"
        />
      </styled.div>
      <styled.div>
        Sign up with Email
      </styled.div>
      <styled.div>
        <hr
          className="full-line"
        />
      </styled.div>
      <styled.div>
        Already have an Airbnb account?
        <span>
          Log in
        </span>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
